dvc/repo/artifacts.py

```diff
@@ -85,3 +107,209 @@ def add(self, name: str, artifact: Artifact, dvcfile: Optional[str] = None):
+        force: bool = False,
+        jobs: Optional[int] = None,
+        dvc_studio_config: Optional[Dict[str, Any]] = None,
+        **kwargs,
+    ) -> Tuple[int, str]:
+        from dvc_studio_client.model_registry import get_download_uris
+
+        from dvc.fs import Callback, HTTPFileSystem, generic, localfs
+        from dvc.utils.studio import env_to_config
+
+        logger.debug("Trying to download artifact '%s' via studio", name)
+        out = out or os.getcwd()
+        to_infos: List[str] = []
+        from_infos: List[str] = []
+        if dvc_studio_config is None:
+            dvc_studio_config = {}
+        dvc_studio_config.update(env_to_config(dict(os.environ)))
+        dvc_studio_config["repo_url"] = repo_url
+        try:
+            for path, url in get_download_uris(
+                repo_url,
+                name,
+                version=version,
+                stage=stage,
+                dvc_studio_config=dvc_studio_config,
+                **kwargs,
+            ).items():
+                to_info = localfs.path.join(out, path)
+                if localfs.exists(to_info) and not force:
+                    hint = "\nTo override it, re-run with '--force'."
+                    raise FileExistsLocallyError(relpath(to_info), hint=hint)
+                to_infos.append(to_info)
+                from_infos.append(url)
+        except ValueError as exc:
+            raise DvcException(
+                f"Failed to download artifact '{name}' via Studio"
+            ) from exc
+        fs = HTTPFileSystem()
+        jobs = jobs or fs.jobs
+        with Callback.as_tqdm_callback(
+            desc=f"Downloading '{name}' from '{repo_url}'",
+            unit="files",
+        ) as cb:
+            cb.set_size(len(from_infos))
+            generic.copy(
+                fs, from_infos, localfs, to_infos, callback=cb, batch_size=jobs
+            )
+
+        return len(to_infos), relpath(localfs.path.commonpath(to_infos))
+
+    @classmethod
+    def get(  # noqa: C901
+        cls,
+        url: str,
+        name: str,
+        version: Optional[str] = None,
+        stage: Optional[str] = None,
+        config: Optional[Union[str, Dict[str, Any]]] = None,
+        out: Optional[str] = None,
+        force: bool = False,
+        jobs: Optional[int] = None,
+    ):
+        from dvc.config import Config
+        from dvc.env import DVC_STUDIO_TOKEN
+        from dvc.repo import Repo
+
+        if version and stage:
+            raise InvalidArgumentError(
+                "Artifact version and stage are mutually exclusive."
+            )
+
+        # NOTE: We try to download the artifact up to three times
+        # 1. via studio with studio config loaded from environment
+        # 2. via studio with studio config loaded from DVC repo 'studio'
+        #    section + environment
+        # 3. via DVC remote
+
+        saved_exc: Optional[Exception] = None
+        if DVC_STUDIO_TOKEN in os.environ:
```

Comment:
Yes, let's drop this condition. It will fail almost immediately when it tries to load the config anyway. This is error-prone because there's some messy logic in how this works now (like reading from both `STUDIO_TOKEN` and `DVC_STUDIO_TOKEN`).